backend/homework/7/src/main/java/org/kdu/service/impl/TexesVehicleServiceFactory.java

```diff
@@ -0,0 +1,94 @@
+package org.kdu.service.impl;
+
+import jakarta.annotation.PostConstruct;
+import org.kdu.constansts.LoggingSystem;
+import org.kdu.entites.Speaker;
+import org.kdu.entites.Tyre;
+import org.kdu.entites.Vehicle;
+import org.kdu.repository.VehicleInventory;
+import org.kdu.service.IVehicleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import java.util.Random;
+
+@Component
+public class TexesVehicleServiceFactory implements IVehicleService {
+
+    Random random = new Random();
+
+
+    String location;
+
+    // injected automatically
+    TyreService tyreService;
+
+    SpeakerService speakerService;
+
+    public VehicleInventory getInventory() {
+        return inventory;
+    }
+
+    // field injection
```

Comment:
this was mentioned in the exercise to showcase field injection so because of that we can see one code smell so please ignore it!